### PR TITLE
deps: unify pest dependency

### PR DESCRIPTION
**Pull request by happysalada**
meilisearch dependends on two different versions of pest.
This can be problematic for some build systems (e.g. NixOS).
Since the repo hasn't received an update in a while, in the meantime, use the later version of the two pest dependencies.

Context: this has been discussed previously https://github.com/meilisearch/MeiliSearch/issues/1273
meilisearch has been selected by ngi to be packaged for nixos. A patch can be applied to make the changes proposed in this PR. This PR intends to see how the maintainers of meilisearch would feel about the patch.

What was done.
- Add an override for the pest dependency in Cargo.toml.
- recreate the Cargo.lock with `cargo update`. This has had the side effect of updating some dependencies.

I ran the tests on darwin. My machine is quite old so I had 8 failures due to a timeout. None of the failures look like they are due to the new dependencies.

Checking the pest repo, it seems there are some recent commits, however no sure date of when there could be a new release.

If this gets accepted, there is no need to do a new release, nixos can just target the new commit.

If you feel it's too much pain for not enough gain, no worries at all!

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [workspace]
-members = [
-    "meilisearch-http",
-    "meilisearch-error",
-]
+members = ["meilisearch-http", "meilisearch-error"]
```
Comment (happysalada) [inline Cargo.toml:2]: This change is unrelated, I'm using an automatic toml formatter and this changed was made. Happy to revert if this proves to be a problem.

File: Cargo.toml [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 
 [profile.release]
 debug = true
+
+[patch.crates-io]
+pest = { git = "https://github.com/pest-parser/pest.git", rev = "51fd1d49f1041f7839975664ef71fe15c7dcaf67" }
```
Comment (happysalada) [inline Cargo.toml:8]: this is the main change of the PR.
The Cargo.toml has just been recreated to take this change into account.

**Comment by happysalada** [comment]
I've reverted the dependency updates so as to not make it confusing to review.

This includes setting the "resolver" version of cargo to 2. With this using `--no-default-features` can also remove default features of workspace crates. Without this, it's not possible to remove default features for `meilisearch-http`

**Comment by MarinPostma** [comment]
Sorry I need a bit of time to look into this. It looks ok, i just want to take a bit of time to understand the fix

**Comment by happysalada** [comment]
No problem at all, thanks a lot for taking a look at this!
I've find a way around setting the resolver version to 2, so the only thing needed is the pest dependency override.
I've applied the patch to the version being packaged in NixOS and updated it to 0.21.1.
No rush.

**Review by MarinPostma — DISMISSED** [review]
Ok this is a good fix. Ultimately, I think we should get rid of the pest dependency, I am currently looking into nom, which I think will be a better long term solution.

thanks!

bors merge

**Comment by happysalada** [comment]
Thanks a lot for merging this, it will make the next release of meilisearch easier to package with!

**Comment by curquiza** [comment]
Needs to have the changes on `release-v0.22.0` to be merge, will be merge on v0.22.0 is out :)

**Comment by happysalada** [comment]
@curquiza let me know if that can be merged now that 0.22 is out.

**Comment by curquiza** [comment]
can be merged!
sorry for the delay

bors merge

**Comment by curquiza** [comment]
bors try

**Comment by curquiza** [comment]
@happysalada looks like there is an issue with your PR with the cargo.lock, cf https://github.com/meilisearch/MeiliSearch/runs/3600376194

**Comment by MarinPostma** [comment]
hey @happysalada can you rebase and then run `cargo update`, I think it should fix the error. Thanks :)

**Comment by happysalada** [comment]
Just rebased off main. Let me know if anything.

**Comment by MarinPostma** [comment]
We'll try

Bors merge

**Review by MarinPostma — APPROVED** [review]
